examples/notebooks/solver_selection.ipynb

```diff
@@ -0,0 +1,245 @@
+    "        \"Voltage [V]\": values[\"Voltage [V]\"].data,\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "# Parameters\n",
+    "parameters = pybop.Parameters(\n",
+    "    pybop.Parameter(\n",
+    "        \"Negative electrode active material volume fraction\",\n",
+    "        prior=pybop.Gaussian(0.6, 0.02),\n",
+    "        bounds=[0.5, 0.8],\n",
+    "    ),\n",
+    "    pybop.Parameter(\n",
+    "        \"Positive electrode active material volume fraction\",\n",
+    "        prior=pybop.Gaussian(0.48, 0.02),\n",
+    "        bounds=[0.4, 0.7],\n",
+    "    ),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "n4OHa-aF04qA"
+   },
+   "source": [
+    "### Defining the solvers for benchmarking\n",
+    "\n",
+    "Now that we have set up the majority of the pybop objects, we construct the solvers we want to benchmark on the given model, and applied current."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "solvers = [\n",
+    "    pybamm.IDAKLUSolver(atol=1e-6, rtol=1e-6),\n",
+    "    pybamm.CasadiSolver(atol=1e-6, rtol=1e-6, mode=\"safe\"),\n",
+    "    pybamm.CasadiSolver(atol=1e-6, rtol=1e-6, mode=\"fast\"),\n",
+    "    pybamm.CasadiSolver(atol=1e-6, rtol=1e-6, mode=\"fast with events\"),\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we construct a range of inputs for the parameters defined above, and select the number of instances in that range to benchmark on. For more statistically repeatable results, increase the variable `n` below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n = 50  # Number of solves\n",
+    "inputs = list(zip(np.linspace(0.45, 0.6, n), np.linspace(0.45, 0.6, n)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, let's benchmark the solvers without sensitivities. This provides a reference for the non-gradient based pybop optimisers and samplers. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Time Evaluate IDA KLU solver: 0.521\n",
+      "Time Evaluate CasADi solver with 'safe' mode: 0.464\n",
+      "Time Evaluate CasADi solver with 'fast' mode: 0.458\n",
+      "Time Evaluate CasADi solver with 'fast with events' mode: 0.451\n"
+     ]
+    }
+   ],
+   "source": [
+    "for solver in solvers:\n",
+    "    model.solver = solver\n",
+    "    problem = pybop.FittingProblem(model, parameters, dataset)\n",
+    "\n",
+    "    start_time = time.time()\n",
+    "    for input_values in inputs:\n",
+    "        problem.evaluate(inputs=input_values)\n",
+    "    print(f\"Time Evaluate {solver.name}: {time.time() - start_time:.3f}\")"
```

Comment:
I think another way to do this could be to use the helper [`pybamm.Timer()`](https://docs.pybamm.org/en/stable/source/api/util.html#pybamm.Timer), it's slightly more tied in with PyBaMM's functionality, though it does use `time.time()` under the hood.

Reply:
Thanks @agriyakhetarpal - I'll keep that in mind for future timings!